File: Partition Array According to Given Pivot.cpp
class Solution {
public:
    vector<int> pivotArray(vector<int>& nums, int pivot) {
        int n = nums.size();
        int less = 0, equal = 0;
        for (const auto& num: nums) {
            if (num < pivot) {
                less++;
            } else if (num == pivot) {
                equal++;
            }
        }
        int i = 0, j = less, k = less + equal;
        vector<int> res(n);
        for (const auto& num: nums) {
            if (num < pivot) {
                res[i++] = num;
            } else if (num == pivot) {
                res[j++] = num;
            } else {
                res[k++] = num;
            }
        }
        return res;
    }
};

File: Partition Array According to Given Pivot/Partition Array According to Given Pivot.java
// Solution for Partition Array According to Given Pivot in JAVA

class Solution {
    public int[] pivotArray(int[] nums, int pivot) {
        int n = nums.length;
        int[] result = new int[n];

        int left = 0, right = n - 1;
        int i = 0, j = n - 1;

        while (i < n && j >= 0) {
            if (nums[i] < pivot) {
                result[left++] = nums[i];
            }

            if (nums[j] > pivot) {
                result[right--] = nums[j];
            }

            i++;
            j--;
        }

        while (left <= right) {
            result[left++] = pivot;
        }

        return result;
    }
}
